src/screens/ExchangeModal.tsx

```diff
@@ -303,6 +266,44 @@ export default function ExchangeModal({
     };
   }, [crosschainSwapsEnabled, inputCurrency?.type, outputCurrency?.type]);
 
+  // if the default input is on a different network than
+  // we want to update the output to be on the same, if its not available -> null
+  const defaultOutputAssetOverride = useMemo(() => {
+    const newOutput = defaultOutputAsset;
+
+    if (
+      defaultInputAsset &&
+      defaultOutputAsset &&
+      defaultInputAsset.type !== defaultOutputAsset.type
+    ) {
+      const crosschainImplementation =
```

Comment:
very minor, feels like `crosschainImplementationAddress` is more accurate?

Reply:
also minor, https://github.com/rainbow-me/rainbow/pull/4281/files#diff-bcc39e0a9afb8c2146a721fb287948c2c1835246c62b2509304194f144308082L234 seems like is the same code than before crosschain i got confused that this was only for crosschain swaps